cpp/include/cudf/strings/string_view.cuh

```diff
@@ -37,47 +57,28 @@ using char_utf8 = uint32_t;  ///< UTF-8 characters are 1-4 bytes
  *
  * The caller must maintain the device memory for the lifetime of this instance.
  *
- * It provides a simple wrapper and string operations for an individual string
- * within a column of strings.
+ * This may be used to wrap a device pointer and size but any member function
+ * that requires accessing the device memory must be called from a kernel.
  */
 class string_view {
  public:
-  /**
-   * @brief Default constructor represents an empty string.
-   */
-  __host__ __device__ string_view();
-
-  /**
-   * @brief Create instance from existing device char array.
-   *
-   * @param data Device char array encoded in UTF8.
-   * @param bytes Number of bytes in data array.
-   */
-  __host__ __device__ string_view(const char* data, size_type bytes);
-
-  string_view(const string_view&) = default;
-  string_view(string_view&&)      = default;
-  ~string_view()                  = default;
-  string_view& operator=(const string_view&) = default;
-  string_view& operator=(string_view&&) = default;
-
   /**
    * @brief Return the number of bytes in this string
    */
-  __host__ __device__ size_type size_bytes() const;
+  __host__ __device__ inline size_type size_bytes() const { return _bytes; }
   /**
    * @brief Return the number of characters in this string
    */
   __device__ size_type length() const;
   /**
    * @brief Return a pointer to the internal device array
    */
-  __host__ __device__ const char* data() const;
+  __host__ __device__ inline const char* data() const { return _data; }
 
   /**
    * @brief Return true if string has no characters
    */
-  __host__ __device__ bool empty() const;
+  __host__ __device__ inline bool empty() const { return size_bytes() == 0; }
```

Comment:
These should use `CUDA_HOST_DEVICE_CALLABLE` so the host/device decorators are removed in `.cpp` files.